components/ntask/org.wso2.carbon.ntask.core/src/main/java/org/wso2/carbon/ntask/core/impl/AbstractQuartzTaskManager.java

```diff
@@ -183,29 +202,50 @@ protected synchronized void scheduleAllTasks() throws TaskException {
         }
     }
 
+    /**
+     * Method to schedule a task that does not require recovery.
+     * @param taskName the name of the task to be scheduled
+     * @throws TaskException if there is an error in scheduling the task
+     */
     protected synchronized void scheduleLocalTask(String taskName) throws TaskException {
-        boolean paused = TaskUtils.isTaskPaused(this.getTaskRepository(), taskName);
-        this.scheduleLocalTask(taskName, paused);
+        scheduleLocalTask(taskName, false);
     }
 
+    /**
+     * Method to schedule a task with whether or not the task is paused and/or requires recovery set to the specified
+     * values.
+     * @param taskName the name of the task to be scheduled
+     * @param paused whether or not the task is paused
+     * @param requestRecovery whether or not the task requires recovery
+     * @throws TaskException if there is an error in scheduling the task
+     */
     protected synchronized void scheduleLocalTask(String taskName,
-                                                  boolean paused) throws TaskException {
+                                                  boolean paused,
```

Comment:
Can't we remove this parameter and do the lookup inside the method using `TaskUtils.isTaskPaused(this.getTaskRepository(), taskName);`?